### PR TITLE
Refactoring of fixed asset transition implementation

### PR DESCRIPTION
**Pull request by piotaixr**
Depends on:
- https://github.com/ekylibre/ekylibre/pull/2517 ✔️ 
- https://github.com/ekylibre/ekylibre/pull/2522 ✔️

### CODE REVIEW

File: app/services/accountancy_duration.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+class AccountancyDuration
```
Comment (Aquaj) [inline app/services/accountancy_duration.rb]: This is a concept inherent to the app, why not simply put it in the models?

File: app/services/accountancy_period.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class AccountancyPeriod
```
Comment (Aquaj) [inline app/services/accountancy_period.rb]: Same as `AccountancyDuration`, couldn't this be simply a model?

File: app/services/fixed_asset/transitions/sell.rb [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+        def update_depreciation_out_on!(out_on)
+          depreciation_out_on = resource.current_depreciation(out_on)
+          return false if depreciation_out_on.nil?
+
+          # check if depreciation have journal_entry
+          if depreciation_out_on.journal_entry
+            raise StandardError, "This fixed asset depreciation is already bookkeep ( Entry : #{depreciation_out_on.journal_entry.number})"
+          end
+
+          next_depreciations = resource.depreciations.where('position > ?', depreciation_out_on.position)
```
Comment (Zank94) [inline app/services/fixed_asset/transitions/sell.rb]: Time to use the `following` method :)

File: app/services/fixed_asset/transitions/start_up.rb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        super && resource.valid? &&
+          during_or_before_opened_financial_year?(resource.started_on)
+      end
+
+      private
+
+        def during_or_before_opened_financial_year?(date)
+          opened_fys = FinancialYear.opened
+
+          opened_fys.count > 0 && date <= opened_fys.last.stopped_on
```
Comment (Zank94) [inline app/services/fixed_asset/transitions/start_up.rb]: `opened_fys.any?` sounds good too

**Pull request merged**